icu4c/source/i18n/measunit_impl.h

```diff
@@ -251,9 +252,7 @@ class U_I18N_API_CLASS MeasureUnitImpl : public UMemory {
      */
     static inline MeasureUnitImpl forCurrencyCode(StringPiece currencyCode) {
         MeasureUnitImpl result;
-        UErrorCode localStatus = U_ZERO_ERROR;
-        result.identifier.append(currencyCode, localStatus);
-        // localStatus is not expected to fail since currencyCode should be 3 chars long
+        result.identifier = currencyCode;
```

Comment:
the old code couldn't fail with a valid currencyCode because the identifier would always store it internally; with a FixedString that always heap-allocates, we get a new OOM failure mode

Reply:
Sure, but that failure would have been outside of the constructor, and the constructor would not have been called. The problem is that the constructor does not have a UErrorCode param, so when it fails, it can't report it.

Reply:
That is totally obscure. Please either add a comment to that effect here, or, better, change this function to take a UErrorCode.